### PR TITLE
Misc copy edits to data-flows sprint 1

### PR DESCRIPTION
**Pull request by illicitonion**
## What does this change?

* Fix up a bunch of the HTML and JS to actually be consistent across steps we claim are consistent.
* Some minor rewordings.

### Common Content?

Yes - edits much of JS3, and adds a new block.

### Org Content?

Sprint

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.MD)
- [x] I have checked my spelling and grammar with an [automated tool](https://www.grammarly.com/grammar-check)
- [x] I have previewed my changes to check the [markdown renders](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) as I intend
- [x] I have run my code to check it works
- [x] My changes follow our [Style Guide](https://curriculum.codeyourfuture.io/guides/code-style-guide)

### CODE REVIEW

File: common-content/en/module/js3/template-html/index.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
 +++
-title = '🍱 Simplifying element creation'
+title = '🍱 Simplifying element creation with <template> tags'
```
Comment (SallyMcGrath) [inline common-content/en/module/js3/template-html/index.md:2]: again here -- titles are too long. 

Really don't want titles longer than 35ch for readiablity. But 55, 60ch will get chomped in search results too!

File: common-content/en/module/js3/template-html/index.md [hunk 2]
```diff
@@ -31,12 +32,12 @@ When we copy an element, its children get copied. This means we can write our te
...
     <p data-director>Director</p>
     <time>Duration</time>
-    <p data-certificate>Certificate</p>
+    <data data-certificate>Certificate</data>
   </section>
 </template>
 ```
 
-This is our template card. Place it in the body of your html. It doesn't show up! [Template HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) is like a wireframe; it's just a _plan_. We can use this template to create a card for any film object. We will clone (copy) this template and populate it with data.
+This is our template card. Place it in the body of your html. It doesn't show up! [Template HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) is like a wireframe; it's just a _plan_. We can use this template to create a card for any film object. We will clone (copy) this template and populate it with data. Replace the contents of your `<script>` tag with this:
```
Comment (SallyMcGrath) [inline common-content/en/module/js3/template-html/index.md:40]: V good note to action

File: org-cyf-itp/content/data-flows/sprints/1/prep/index.md [hunk 3]
```diff
@@ -5,33 +5,36 @@ emoji= '🧑🏾‍💻'
...
-name="composing elements"
+name="Composing elements"
 src="module/js3/composing-elements"
 [[blocks]]
-name="Simplifying element creation"
+name="Simplifying element creation with functions"
 src="module/js3/simplifying-element-creation"
 [[blocks]]
-name="<template>"
+name="Simplifying element creation with <template> tags"
```
Comment (SallyMcGrath) [inline org-cyf-itp/content/data-flows/sprints/1/prep/index.md:26]: You're a title maniac! 😉

**Review by SallyMcGrath — CHANGES_REQUESTED** [review]
LGTM except the titles are too long.

innerText reflows the page, I think. I don't think it matters in this tiny use case, so if you want it, have it.

File: common-content/en/module/js3/single-datum/index.md [hunk 4]
```diff
@@ -1,9 +1,10 @@
 +++
-title = '💽 Single datum'
+title = '💽 Rendering a single datum'
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply common-content/en/module/js3/single-datum/index.md:2]: So... My motivation here is that "Single datum" is super unhelpful on a page like https://programming.codeyourfuture.io/overview/ (and not all that helpful on the prep view either)...

I suppose we could consciously use the sprint view names and the frontmatter titles differently in different contexts?

WDYT?

File: common-content/en/module/js3/template-html/index.md (hunk 1 above)
Comment (illicitonion) [reply common-content/en/module/js3/template-html/index.md:2]: Similarly here, having two consecutive blocks in the overview which are both titled "Simplifying element creation" seems unhelpful...

File: common-content/en/module/js3/components/index.md [hunk 5]
```diff
@@ -31,10 +33,10 @@ const film = {
 const createFilmCard = (template, film) => {
   const card = template.content.cloneNode(true);
   // Now we are querying our cloned fragment, not the entire page.
-  card.querySelector("h3").textContent = film.title;
-  card.querySelector("p").textContent = `Director: ${film.director}`;
-  card.querySelector("time").textContent = `${film.duration} minutes`;
-  card.querySelector("data").textContent = `Certificate: ${film.certificate}`;
+  card.querySelector("h3").innerText = film.title;
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply common-content/en/module/js3/components/index.md:36]: Ooh TIL! I will go update everything else to `textContent`.

File: common-content/en/module/js3/composing-elements/index.md [hunk 6]
```diff
@@ -30,13 +30,19 @@ const film = {
...
 If we open up the console tab, we should be able to see this element logged in the console. However, it won't yet appear in the browser.
 
-{{<note type="tip" title="Uncaught ReferenceError: document is not defined">}} If you see this error, make sure you are running your code in the browser and not VSCode. Node doesn't have the DOM API. You need to use your browser console. See [how to set up your html](#rendering-from-data-1) if you are stuck.{{</note>}}
+{{<note type="tip">}} If you see this error:
+
+```
+Uncaught ReferenceError: document is not defined
+```
+
+make sure you are running your code in the browser and not a terminal. Node doesn't have the DOM API. You need to use your browser console. See [how to set up your html](#rendering-from-data-1) if you are stuck.{{</note>}}
```
[the remark this answers is not in the document]
Comment (illicitonion) [reply common-content/en/module/js3/composing-elements/index.md:45]: This is a link to a tab within the page.

I don't love it as-is, and I'm happy to remove it and instead say "See the HTML tab" if you think that's more clear?

[the remark this answers is not in the document]
Comment (SallyMcGrath) [reply common-content/en/module/js3/composing-elements/index.md:45]: I don't know if it is more clear 🤔 It's up to you -- I'm no help here.

File: common-content/en/module/js3/template-html/index.md (hunk 1 above)
Comment (SallyMcGrath) [reply common-content/en/module/js3/template-html/index.md:2]: How about 

Creating elements with <template> ?

File: common-content/en/module/js3/single-datum/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (SallyMcGrath) [reply common-content/en/module/js3/single-datum/index.md:2]: I'd move the emojis to the emoji field and call "single datum" "render one card"

File: common-content/en/module/js3/simplifying-element-creation/index.md [hunk 7]
```diff
@@ -30,10 +31,10 @@ We can identify things we're doing several times, and extract a function to do t
 In this example, we keep doing these three things:
 
 1. Create a new element (sometimes with a different tag name).
-2. Set that element's text content (always to different values).
+2. Set that element's inner text (always to different values).
```
Comment (SallyMcGrath) [inline common-content/en/module/js3/simplifying-element-creation/index.md:34]:
```suggestion
2. Set that element's text content (always to different values).
```

You changed it back remember ;)

**Review by SallyMcGrath — APPROVED** [review]
one note, beautiful work thank you!

**Pull request merged**